src/git-repository.coffee

```diff
@@ -79,6 +79,9 @@ class GitRepository
     unless @repo?
       throw new Error("No Git repository found searching path: #{path}")
 
+    @watch = fs.watch path, { persistent: false, recursive: true }, (event, file) =>
```

Comment:
I tried out chokidar (and alternatives, like watchr) but they won't solve the issue when there are directories with a large amount of files (e.g. `atom/atom` repository).

There seems to be a pull request for Windows support (https://github.com/joyent/libuv/pull/1473) but it has to be merged yet (and Linux would still be a problem; even the new [fanotify](http://man7.org/linux/man-pages/man7/fanotify.7.html) doesn't work out of the box). This is quite sad and I assumed this problem was something which operating systems had already solved long time ago.

Constantly polling the git repository could be a solution at this point: we may setup a timer which runs, for instance, every second and fetches updates from the repository. Other than removing the need of subscribing to window focus and to buffer change, it would still solve #5301.

@kevinsawicki: my last commit is the exact implementation of what I said above. It works quite well on my machine and it doesn't eat a lot of resources.

Meanwhile, I'll change the name of the PR as well, given that we've discarded the `fs.watch` option.